test/suite/hook_tiered.py

```diff
@@ -52,35 +60,51 @@
 
 # These are the hook functions that are run when particular APIs are called.
 
+# Add the local storage extension whenever we call wiredtiger_open
+def wiredtiger_open_tiered(ignored_self, args):
+    auth_token = "test_token"
+    bucket = "mybucket"
+    extension_name = "local_store"
+    prefix = "pfx-"
+    # XXX This assumes we are running from WT_TEST/test* in the build directory!
+    extension_library = os.getcwd() + \
+            '/../../ext/storage_sources/local_store/.libs/libwiredtiger_local_store.so'
```

Comment:
This works, but is obviously ugly.  Is there state available from the hook that we could use to find the build dir or the root of the tree? This would let us find the extension in a more general way.

Reply:
Looking at `test_encrypt01.py` seems to indicate that defining a `conn_extensions` function "does the right thing" and that uses `extlist.extension()` to add them.

Reply:
Yes, unfortunately I think that code is all predicated on having a WiredTigerTestCase object, and that we don't have.  I would do this to get the build directory.  In `wttest.py`, somewhere in `class WiredTigerTestCase` add:
```
    @staticmethod
    def buildDirectory():
        return WiredTigerTestCase._builddir
```
Then from the hooks you can:
```
from wttest import WiredTigerTestCase
...
      builddir = `WiredTigerTestCase.buildDirectory()
```
If you want to do this more cleanly, you could probably do some refactoring in `wttest.py` that broke out some code in `extensionConfig` into a static function that could be called from hooks.  The part that "knows" about the build directory and how to add `./ext/XXX/YYY/.libs/libwiredtiger_YYY.so`

Reply:
I refactored `WiredTigerTestCase.extensionsConfig` as suggested.  Thanks for the suggestion!

Reply:
Great, your changes for that look good.

Reply:
Thanks Don. I was going to ask you to re-review, but you beat me to it!